base/strings/types.jl

```diff
@@ -92,24 +92,30 @@ isvalid(s::SubString{<:DirectIndexString}, i::Integer) = (start(s) <= i <= endof
 ind2chr(s::SubString{<:DirectIndexString}, i::Integer) = begin checkbounds(s,i); i end
 chr2ind(s::SubString{<:DirectIndexString}, i::Integer) = begin checkbounds(s,i); i end
 
+# need to define nextind and prevind only for SubString{String}
+# as other cases are handled by definitions for AbstractString
 function nextind(s::SubString{String}, i::Integer)
+    # handle the case when i+s.offset is greater than 1
     i < 1 && return 1
     nextind(s.string, i+s.offset)-s.offset
 end
 
 function prevind(s::SubString{String}, i::Integer)
     e = endof(s)
+    # handle the case when i+s.offset is greater than endof(s)
     i > e && return e
     prevind(s.string, i+s.offset)-s.offset
 end
 
 function nextind(s::SubString{String}, i::Integer, nchar::Integer)
+    # nextind for nchar=1 must return nonnegative value
```

Comment:
I don't understand why this would apply only to "nchar=1".

Reply:
rewritten, hopefully it is clearer now